client/src/pages/Main.tsx

```diff
@@ -34,7 +35,17 @@ const Main: React.FC<MainProps> = ({ messageCount, messagePosition, messageDisap
             }, messageDisappearTime);
             setMessageTimeouts(prev => ({ ...prev, [msgId]: timeoutId }));
         };
-    }, [messageDisappearTime]);
+
+        eventSource.addEventListener('message', handleMessageEvent);
+
+        return () => {
+            eventSource.removeEventListener('message', handleMessageEvent);
+
+            Object.values(messageTimeouts).forEach(timeoutId => {
+                clearTimeout(timeoutId);
+            });
+        };
+    }, [eventSource, messageDisappearTime]);
```

Comment:
EventListeners for Main are added when Main is mounted (i.e. go to Main page), and removed when Main is unmounted (i.e. go to Setting page). Timeouts are cleared when Main component unmounts.